src/linux_ip_socket/linux_ip_socket.cpp

```diff
@@ -0,0 +1,336 @@
+ * Version 2.3 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://essr.esa.int/license/list
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "linux_ip_socket.h"
+
+#include <cstdint>
+#include <cstdlib>
+#include <cstring>
+#include <iostream>
+#include <string>
+
+#include <arpa/inet.h>
+#include <netdb.h>
+#include <netinet/in.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <errno.h>
+#include <poll.h>
+#include <unistd.h>
+
+linux_ip_socket_private_data::linux_ip_socket_private_data()
+    : m_thread(DRIVER_THREAD_PRIORITY, DRIVER_THREAD_STACK_SIZE)
+    , m_message_buffer_index(0)
+{
+}
+
+void
+linux_ip_socket_private_data::driver_init(const SystemBus bus_id,
+                                          const SystemDevice device_id,
+                                          const Socket_IP_Conf_T* const device_configuration,
+                                          const Socket_IP_Conf_T* const remote_device_configuration)
+{
+    m_ip_device_bus_id = bus_id;
+    m_ip_device_id = device_id;
+    m_ip_device_configuration = device_configuration;
+    m_ip_remote_device_configuration = remote_device_configuration;
+    m_thread.start(&taste::LinuxIpSocketPoll, this);
+}
+
+void
+linux_ip_socket_private_data::driver_poll()
+{
+    prepare_listen_socket();
+
+    const int descriptors_table_size = DRIVER_MAX_CONNECTIONS + 1;
+    struct pollfd descriptors_table[descriptors_table_size];
+    int descriptors_table_count = 0;
+    descriptors_table[0].fd = m_listen_sockfd;
+    descriptors_table[0].events = POLLIN;
+    ++descriptors_table_count;
+
+    while(true) {
+        const int poll_result = ::poll(descriptors_table, descriptors_table_count, -1);
+        if(poll_result == -1) {
+            std::cerr << "poll() returned an error: " << strerror(errno) << std::endl;
+            std::cerr << "aborting." << std::endl;
+            abort();
+        }
+
+        for(int descriptor_index = 0; descriptor_index < descriptors_table_count; ++descriptor_index) {
+            if(descriptors_table[descriptor_index].revents & POLLIN) {
+                if(descriptor_index == 0) {
+                    accept_connection(descriptors_table, descriptors_table_count);
+                } else {
+                    read_data_or_disconnect(descriptor_index, descriptors_table, descriptors_table_count);
+                }
+            }
+        }
+    }
+}
+
+void
+linux_ip_socket_private_data::driver_send(const uint8_t* const data, const size_t length)
+{
+    const int sockfd = connect_to_remote_driver();
```

Comment:
Are we connecting and disconnecting for each send? Won't this cause issues when sending messages rapidly?

Reply:
To be evaluated. There's possibility that it may cause issues.